File: life-assisitant-api/internal/service/notification.go
// Package service 通知服务接口
package service

import (
	"context"

	"github.com/life-assistant/api/internal/model/dto"
)

// INotificationService 通知服务
// 落库通知的列表 / 未读数 / 标记已读
type INotificationService interface {
	// List 通知列表（分页，created_at 倒序）
	List(ctx context.Context, req *dto.ListNotificationsReq) (*dto.ListNotificationsResp, error)

	// UnreadCount 获取当前用户未读通知数
	UnreadCount(ctx context.Context) (*dto.UnreadCountResp, error)

	// MarkRead 标记所有通知为已读，返回受影响条数
	MarkRead(ctx context.Context) (*dto.MarkReadResp, error)

	// MarkReadByID 标记单条通知为已读
	MarkReadByID(ctx context.Context, id string) error
}

File: life-assisitant-ui-mobile/src/api/notification.ts
/**
 * 通知 API
 * 路径前缀：/notifications
 *
 * SYNC-FROM-BACKEND: life-assisitant-api/internal/controller/notification.go
 * 最后同步：2026-09-18
 *
 * 后端端点：
 *   GET  /notifications                 notification:view   列表（response.Page）
 *   GET  /notifications/unread-count    notification:view   未读数
 *   POST /notifications/mark-read       notification:handle 全部已读
 *   POST /notifications/:id/read        notification:handle 单条已读
 */
import { http } from './request'
import type {
  ListNotificationsResp,
  MarkReadResp,
  PageQuery,
  UnreadCountResp,
} from './types'

export const notificationApi = {
  /** 通知列表（GET /notifications，走 response.Page 含分页四件套） */
  list(params: PageQuery = {}): Promise<ListNotificationsResp> {
    return http.get<ListNotificationsResp>('/notifications', { params })
  },

  /** 未读数量（GET /notifications/unread-count） */
  unreadCount(): Promise<UnreadCountResp> {
    return http.get<UnreadCountResp>('/notifications/unread-count')
  },

  /** 全部标记已读（POST /notifications/mark-read） */
  markRead(): Promise<MarkReadResp> {
    return http.post<MarkReadResp>('/notifications/mark-read')
  },

  /** 单条标记已读（POST /notifications/:id/read） */
  markReadByID(id: string): Promise<MarkReadResp> {
    return http.post<MarkReadResp>(`/notifications/${id}/read`)
  },
}
